File: dart/lib/model/v3_outlet_response.dart
part of swagger.api;

class V3OutletResponse {
  /* myki ticket outlets */
  List<V3Outlet> outlets = [];

  V3Status status = null;

  V3OutletResponse();

  @override
  String toString() {
    return 'V3OutletResponse[outlets=$outlets, status=$status, ]';
  }

  V3OutletResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    outlets = V3Outlet.listFromJson(json['outlets']);
    status = new V3Status.fromJson(json['status']);
  }

  Map<String, dynamic> toJson() {
    return {
      'outlets': outlets,
      'status': status
     };
  }

  static List<V3OutletResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3OutletResponse>() : json.map((value) => new V3OutletResponse.fromJson(value)).toList();
  }

  static Map<String, V3OutletResponse> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3OutletResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3OutletResponse.fromJson(value));
    }
    return map;
  }
}

File: java-vertx/src/main/java/io/swagger/server/api/model/V3OutletResponse.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.server.api.model.V3Outlet;
import io.swagger.server.api.model.V3Status;
import java.util.ArrayList;
import java.util.List;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3OutletResponse
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3OutletResponse   {
  
  private List<V3Outlet> outlets = null;

  private V3Status status = null;

  public V3OutletResponse outlets(List<V3Outlet> outlets) {
    this.outlets = outlets;
    return this;
  }

  public V3OutletResponse addOutletsItem(V3Outlet outletsItem) {
    if (this.outlets == null) {
      this.outlets = new ArrayList<V3Outlet>();
    }
    this.outlets.add(outletsItem);
    return this;
  }

  /**
   * myki ticket outlets
   * @return outlets
   **/
    public List<V3Outlet> getOutlets() {
    return outlets;
  }

  public void setOutlets(List<V3Outlet> outlets) {
    this.outlets = outlets;
  }

  public V3OutletResponse status(V3Status status) {
    this.status = status;
    return this;
  }

  /**
   * Get status
   * @return status
   **/
    public V3Status getStatus() {
    return status;
  }

  public void setStatus(V3Status status) {
    this.status = status;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3OutletResponse v3OutletResponse = (V3OutletResponse) o;
    return Objects.equals(this.outlets, v3OutletResponse.outlets) &&
        Objects.equals(this.status, v3OutletResponse.status);
  }

  @Override
  public int hashCode() {
    return Objects.hash(outlets, status);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3OutletResponse {\n");
    
    sb.append("    outlets: ").append(toIndentedString(outlets)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
